pkg/controller/util.go

```diff
@@ -1087,16 +1087,22 @@ func getDefaultAccessModes(c client.Client, storageClass *storagev1.StorageClass
 	return nil, errors.Errorf("no accessMode defined on StorageProfile for %s StorageClass", storageClass.Name)
 }
 
-func volumeSize(c client.Client, storage *cdiv1.StorageSpec, volumeMode *v1.PersistentVolumeMode) (*resource.Quantity, error) {
+func volumeSize(c client.Client, spec cdiv1.DataVolumeSpec, volumeMode *v1.PersistentVolumeMode) (*resource.Quantity, error) {
 	// resources.requests[storage] - just copy it to pvc,
-	requestedSize, found := storage.Resources.Requests[v1.ResourceStorage]
+	requestedSize, found := spec.Storage.Resources.Requests[v1.ResourceStorage]
+
 	if !found {
-		return nil, errors.Errorf("Datavolume Spec is not valid - missing storage size")
+		// Storage size can be empty when using PVC source
+		if spec.Source.PVC != nil {
+			return &requestedSize, nil
```

Comment:
what is the value of `requestedSize` if it is not Found. Isn't it nil?

Reply:
In that case `requestedSize` would be 0, so it can be later checked with the `IsZero()` method. Simply returning `nil` would later cause a nil pointer dereference, so I chose to return the empty `requestedSize`. That said, I'm not 100% happy with that, so I can try to make it more readable.

Reply:
maybe a const somewhere that says ZERO, maybe there is such Quantity const defined somewhere?

Reply:
since this is pretty big PR and my comment is just a style nitpick I'll close this comment as resolved, and it can be done later if anyone thinks it can be improved